test/wdio/computed-properties-watch-decorator/cmp.test.tsx

```diff
@@ -0,0 +1,47 @@
+import { Fragment, h } from '@stencil/core';
+import { render } from '@wdio/browser-runner/stencil';
+
+describe('computed-properties-watch-decorator', function () {
+  beforeEach(async () => {
+    render({
+      template: () => (
+        <>
+          <computed-properties-watch-decorator></computed-properties-watch-decorator>
+          <button type="button">Trigger watch callbacks</button>
+        </>
+      ),
+    });
+    document.querySelector('button').addEventListener('click', () => {
+      const cmp = document.querySelector('computed-properties-watch-decorator');
+      cmp.setAttribute('first', 'Bob');
+      cmp.setAttribute('last', 'Builder');
+    });
+  });
+
+  it('triggers the watch callback when the associated prop changes', async () => {
+    const el = document.querySelector('computed-properties-watch-decorator');
+    await expect(el).toHaveText(['First name called with: not yet', 'Last name called with: not yet'].join('\n'));
```

Comment:
In the [Karma version of the test](
https://github.com/ionic-team/stencil/pull/5456/files#diff-0bc7d13655d1b50de6b97c648eccd3328381c75b048b436f146a54906b065e65L14) (where we didn't use `toHaveText`):
```
    expect(el.innerText).toBe('First name called with: not yet\n\nLast name called with: not yet');
```

I noticed the double `\n` in the old test, whereas we have a single one in this test. I'm guessing this has to do with how `toHaveText` operates (I'm guessing not on `innerText`, perhaps `textContent`?), is that the reason for this (tiny) difference? 

FWIW, the rendered component looks correct to me, which is why I assume this is just an API difference, but I could be wrong!

Reply:
The WebDriver "[get element test](https://www.w3.org/TR/webdriver2/#get-element-text)" command returns a representation of the "visible" state of a text. While it is not possible to define this concretely, they are relying on an implementation for it that has been used for many years. This implementation seems to remove double `\n` as they are not impacting how the text is rendered. `innerText` on the other hand actually gives you the characters of the DOM node which doesn't contains any filtering.